library/src/main/java/com/getstream/sdk/chat/view/MessageInputView.kt

```diff
@@ -0,0 +1,501 @@
+package com.getstream.sdk.chat.view
+
+import android.Manifest
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.os.Bundle
+import android.os.StrictMode
+import android.os.StrictMode.VmPolicy
+import android.text.Editable
+import android.text.TextUtils
+import android.util.AttributeSet
+import android.view.KeyEvent
+import android.view.LayoutInflater
+import android.view.View
+import android.view.View.OnFocusChangeListener
+import android.widget.RelativeLayout
+import androidx.core.os.BuildCompat
+import androidx.core.view.inputmethod.InputConnectionCompat
+import androidx.core.view.inputmethod.InputContentInfoCompat
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.Observer
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.getstream.sdk.chat.Chat.Companion.getInstance
+import com.getstream.sdk.chat.R
+import com.getstream.sdk.chat.databinding.StreamViewMessageInputBinding
+import com.getstream.sdk.chat.enums.MessageInputType
+import com.getstream.sdk.chat.interfaces.MessageSendListener
+import com.getstream.sdk.chat.model.AttachmentMetaData
+import com.getstream.sdk.chat.model.ModelType
+import com.getstream.sdk.chat.navigation.destinations.CameraDestination
+import com.getstream.sdk.chat.utils.CaptureController
+import com.getstream.sdk.chat.utils.Constant
+import com.getstream.sdk.chat.utils.GridSpacingItemDecoration
+import com.getstream.sdk.chat.utils.LlcMigrationUtils
+import com.getstream.sdk.chat.utils.MessageInputController
+import com.getstream.sdk.chat.utils.PermissionChecker
+import com.getstream.sdk.chat.utils.StringUtility
+import com.getstream.sdk.chat.utils.TextViewUtils
+import com.getstream.sdk.chat.utils.Utils
+import com.getstream.sdk.chat.viewmodel.MessageInputViewModel
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.models.Attachment
+import io.getstream.chat.android.client.models.Command
+import io.getstream.chat.android.client.models.Member
+import io.getstream.chat.android.client.models.Message
+import net.yslibrary.android.keyboardvisibilityevent.KeyboardVisibilityEvent
+import java.io.File
+import java.util.Date
+import java.util.UUID
+
+class MessageInputView(context: Context, attrs: AttributeSet?) : RelativeLayout(context, attrs) {
+	/**
+	 * Tag for logging purposes
+	 */
+	val TAG = MessageInputView::class.java.simpleName
+	private val binding: StreamViewMessageInputBinding = StreamViewMessageInputBinding.inflate(LayoutInflater.from(context), this, true)
+
+	/**
+	 * Styling class for the MessageInput
+	 */
+	private val style: MessageInputStyle = MessageInputStyle(context, attrs)
+
+	/**
+	 * Fired when a message is sent
+	 */
+	private var messageSendListener: MessageSendListener? = null
+
+	/**
+	 * Permission Request listener
```

Comment:
Quite useless comments. When field name is just splitted by spaces it's a bad sign.

Reply:
It was done by "Java to Kotlin auto-convert" from the previous `MessageInputView.java`

Reply:
That's new. Ok.
But still if you see excessive comment just delete it. 1 second improvement makes easier future reviews.